### PR TITLE
chore(lint): require semicolons

### LINKED ISSUE #817: relying on ASI (auto semicolon insertion) in AtScript makes tooling harder
Stock Webstorm gives a warning on statements which don't end with semi. We could turn the warning off in the IDE, but I'd rather not rely on ASI.

We'd like to format the codebase with clang-format, but it doesn't understand javascript syntax enough to understand a statement which terminates with a linebreak instead of a semicolon. Instead it removes the linebreak, breaking the code.

In general, it's a pity to carry forward ASI into AtScript since it complicates all the tooling. Does anyone have a strong desire to omit semicolons in AtScript code?

@mprobst suggests this as well. We are working on a mechanism for finding all the ASI abuses in the codebase.

### PR DESCRIPTION
**Pull request by alexeagle**
Relying on ASI (automatic semicolon insertion)
is allowed in TypeScript because JavaScript allows
it. However, when we run clang-format it doesn’t
understand that these statements are terminated
with a newline and changes the indentation, in bad
cases even breaking the code.

Fixes #817

### CODE REVIEW

File: modules/angular2/src/change_detection/pipes/observable_pipe.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export class ObservablePipe implements Pipe {
 
   _subscribe(obs: Observable): void {
     this._observable = obs;
-    this._subscription = ObservableWrapper.subscribe(obs, value => {this._updateLatestValue(value)},
+    this._subscription = ObservableWrapper.subscribe(obs, value => this._updateLatestValue(value),
```
Comment (btford) [inline modules/angular2/src/change_detection/pipes/observable_pipe.ts]: This changes the meaning of the code; does `ObservableWrapper.subscribe` care about the return value of the function argument?

**Comment by btford** [comment]
If the change I commented on is inconsequential, then feel free to merge.

**Comment by alexeagle** [comment]
Yeah, in that case it's extra fine because _updateLatestValue is void so
the function literal is still void

**Comment by alexeagle** [comment]
thanks, fixed the new violations after rebase, and merged as 93055f7

File: modules/angular2/src/change_detection/pipes/observable_pipe.ts (hunk 1 above)
Comment (jeffbcross) [inline modules/angular2/src/change_detection/pipes/observable_pipe.ts]: It doesn't care here.

**Pull request closed without merging**